### PR TITLE
FWI-3400 : Verify rule cli refactoring

### PR DESCRIPTION
**Pull request by jdesouza**
This PR fixes #

## Checklist
* [x] I have signed the CLA
* [x] I have updated/added any relevant documentation

### CODE REVIEW

File: cmd/insights/verify-rule.go [hunk 1]
```diff
@@ -6,51 +6,89 @@ import (
...
-		err = yaml.Unmarshal(inputContents, &verifyRule)
+		err = yaml.Unmarshal(aiBytes, &actionItem)
 		if err != nil {
 			exitWithError("Could not parse input file", err)
 		}
-		result, err := rules.RunVerifyRule(org, insightsToken, host, verifyRule)
+
+		ruleInput, err := os.Open(automationRuleFile)
+		if err != nil {
+			exitWithError(fmt.Sprintf("Error when trying to open file %s", actionItemFile), err)
```
Comment (ivanfetch-fw) [inline cmd/insights/verify-rule.go:49]: This error refers to the incorrect (action item) file name, as does the next error below it.

**Review by ivanfetch-fw — CHANGES_REQUESTED** [review]
James, thank you for the extra work on this, to help automation-rule validation to match the existing CLI style and user experience! I've commented on a few minor fixes.

File: cmd/insights/verify-rule.go [hunk 2]
```diff
@@ -6,51 +6,89 @@ import (
...
-		inputContents, err := io.ReadAll(in)
+		aiBytes, err := io.ReadAll(aiInput)
 		if err != nil {
-			exitWithError(fmt.Sprintf("Could not read file %s", inputFile), err)
+			exitWithError(fmt.Sprintf("Could not read file %s", actionItemFile), err)
 		}
-		err = yaml.Unmarshal(inputContents, &verifyRule)
+		err = yaml.Unmarshal(aiBytes, &actionItem)
 		if err != nil {
 			exitWithError("Could not parse input file", err)
```
[the remark this answers is not in the document]
Comment (jdesouza) [reply cmd/insights/verify-rule.go:44]: done

File: cmd/insights/verify-rule.go (hunk 1 above)
Comment (jdesouza) [reply cmd/insights/verify-rule.go:49]: done

File: cmd/insights/verify-rule.go [hunk 3]
```diff
@@ -6,51 +6,89 @@ import (
 	"os"
```
[the remark this answers is not in the document]
Comment (jdesouza) [reply cmd/insights/verify-rule.go]: done

File: cmd/insights/verify-rule.go [hunk 4]
```diff
@@ -6,51 +6,89 @@ import (
...
+			exitWithError(fmt.Sprintf("Error when trying to open file %s", actionItemFile), err)
+		}
+		ruleBytes, err := io.ReadAll(ruleInput)
+		if err != nil {
+			exitWithError(fmt.Sprintf("Could not read file %s", actionItemFile), err)
+		}
+		verifyRule := rules.VerifyRule{
+			ActionItem: actionItem,
+			Context:    rules.RuleExecutionContext(insightsContext),
+			ReportType: reportType,
```
[the remark this answers is not in the document]
Comment (jdesouza) [reply cmd/insights/verify-rule.go]: done

**Review by ivanfetch-fw — APPROVED** [review]

**Pull request merged**